distributed/client.py

```diff
@@ -3610,18 +3623,15 @@ def track_future(self, future):
         try:
             yield _wait(future)
         except CancelledError:
-            del self.futures[future]
-            if not self.futures:
-                self._notify()
-            return
+            pass
         if self.with_results:
-            result = yield future._result()
+            result = yield future._result(raiseit=False)
         with self.lock:
             self.futures[future] -= 1
             if not self.futures[future]:
                 del self.futures[future]
             if self.with_results:
-                self.queue.put_nowait((future, result))
+                self.queue.put_nowait((future, result, future.status == 'error'))
```

Comment:
Ah, your solution here points to the fact that including the third element of the tuple is unnecessary.  We can compute it from the first.  My apologies for leading you astray.  I would compute the `erred` value just in the `_get_and_raise` method as you've done it here.

Reply:
Ah ok, no worries!